### ISSUE TITLE
Component File Hierarchy Discrepancy between CD Definition and Github Project

### ISSUE THREAD

**Issue by cruzerika** [clearlydefined/service#646]
We discovered that the CD definition for the component Glib does not have the same file hierarchy as the project in Github. There are missing files on the CD definition. 
 
Link to the CD definition: https://clearlydefined.io/definitions/git/github/gnome/glib/99befa31f8f8d02260728a1547832cba37b807e1
Link to the github project: https://github.com/GNOME/glib/tree/99befa31f8f8d02260728a1547832cba37b807e1/docs/reference
 
In particular, the following folders have discrepancies:
• Glib -> docs -> reference 
On the CD definition, the file hierarchy for this folder is missing a couple of files (i.e. AUTHORS, Makefile.am, etc.) compared to what is shown in github
<img width="916" alt="Glib CD definition reference folder" src="https://user-images.githubusercontent.com/44687374/67337053-3f635380-f4db-11e9-93f3-7ed67e54557f.png">

<img width="1070" alt="Glib Github reference folder" src="https://user-images.githubusercontent.com/44687374/67336595-77b66200-f4da-11e9-9916-52b728f814ca.png">

• Glib -> docs -> reference -> glib
On the CD definition, the file hierarchy for this folder is missing a number of files (i.e. image files and a number of xml files) compared to what is shown in github
<img width="1154" alt="Glib CD definition glib folder" src="https://user-images.githubusercontent.com/44687374/67336590-771dcb80-f4da-11e9-864d-85f6c557ffc1.png"> 
 
<img width="1066" alt="Glib Github glib folder" src="https://user-images.githubusercontent.com/44687374/67336593-77b66200-f4da-11e9-92a2-79044b86a582.png">

As well, on the CD definition, the path Glib -> docs -> reference which contains a COPYING file, did not have any license detected by the scan tools. The license field was blank although there is a license text on the COPYING file. I understand that the license text might not exist as an SPDX identifier but shouldn't the tool at least identify the license as NOASSERTION?https://github.com/GNOME/glib/blob/99befa31f8f8d02260728a1547832cba37b807e1/docs/reference/COPYING
<img width="870" alt="Screen Shot 2019-10-22 at 2 52 07 PM" src="https://user-images.githubusercontent.com/44687374/67337315-a7b23500-f4db-11e9-87a3-ac9e6fd1a9ad.png">

 Can anyone please help check the issue? Thank you.

**Comment by geneh** [comment]
This component had old version of the tools run: clearlydefined/1, scancode/2.9.1
After re-harvesting this component the new tools versions are clearlydefined/1.3.0, licensee/9.13.0, scancode/3.2.2.
The missing files are present now.

**Comment by cruzerika** [comment]
Thank you @geneh for looking into this issue and for the quick resolution of the missing files. However, I still see an empty license field for the COPYING file. Would this be an expected behaviour? From my understanding, if a file contains a license text, the license field should have a license and if license is not supported by the SPDX identifier, then it should at least say NOASSERTION. Is my understanding correct?

I also have a question about re-harvesting of components to have the new tool versions run. From my understanding, the re-harvesting is triggered automatically every time there is a new tool version released. It seems like for this specific component, the re-harvesting was not triggered automatically once there was a new version of the tool released. Is my assumption correct? If so, could you kindly confirm why this is the case? 

Thank you.

**Comment by geneh** [comment]
> I also have a question about re-harvesting of components to have the new tool versions run. From my understanding, the re-harvesting is triggered automatically every time there is a new tool version released. It seems like for this specific component, the re-harvesting was not triggered automatically once there was a new version of the tool released. Is my assumption correct? If so, could you kindly confirm why this is the case?

No, the re-harvesting is not triggered automatically every time there is a new tool version released.

**Comment by pombredanne** [comment]
@cruzerika @geneh note that in this case the COPYING also has detection bug in that version (older) version of scancode. It is correctly detected in the latest version though as lgpl-2.0.

**Comment by jeffmendoza** [comment]
@ignacionr Question here about facets.

**Comment by ignacionr** [comment]
Hello @jeffmendoza you mention a question for me. Kindly make it explicit.

**Comment by jeffmendoza** [comment]
@ignacionr Sorry, was doing a bulk triage. Re-reading this here, looks like both questions from https://github.com/clearlydefined/service/issues/646#issuecomment-547183171 have been answered.

**Comment by cruzerika** [comment]
> @cruzerika @geneh note that in this case the COPYING also has detection bug in that version (older) version of scancode. It is correctly detected in the latest version though as lgpl-2.0.

@pombredanne: I think there might be a slight confusion on the COPYING file I am referring to. I am referring to the one found in the path Glib -> docs -> reference. I still see that the license field is empty. Am I missing something?

**Comment by pombredanne** [comment]
@cruzerika re:

> I think there might be a slight confusion on the COPYING file I am referring to. I am referring to the one found in the path Glib -> docs -> reference. I still see that the license field is empty. Am I missing something?

Indeed.. I mixed this up. In that file, scancode-wise there are some things detected and eventually there should be some  OTHER of sorts since there was a detection: it should not be empty

 (on a side note I am also adding a new license detection rule in Scancode for this specific license formulation)

**Comment by cruzerika** [comment]
Thank you, @pombredanne. I would be very interested to know when the new license detection rule will be added into Scancode. Once you got a chance to add it, could you kindly notify me? Thank you.
